docs/source/format/Columnar.rst

```diff
@@ -1019,7 +1024,9 @@ Schematically we have: ::
 In the file format, there is no requirement that dictionary keys
 should be defined in a ``DictionaryBatch`` before they are used in a
 ``RecordBatch``, as long as the keys are defined somewhere in the
-file.
+file. There must be one dictionary batch per dictionary encoded column.
+Even if a all record batches are null for a column an empty dictionary
+batch is expected. Delta dictionaries are not used in the file format.
```

Comment:
@pitrou updated per ML conversation, does this look reasonable to you?

Reply:
To me yes, but it looks like the conversation is not done ;-)